### PR TITLE
refacter_show_course_user

### PR DESCRIPTION
**Pull request by hienbx94**
I have created a pull request. Please check it.

### CODE REVIEW

**Comment by trantube94** [comment]
OK

File: app/controllers/subjects_controller.rb [hunk 1]
```diff
@@ -3,6 +3,17 @@ class SubjectsController < ApplicationController
   before_action :check_status, only: :update
   before_action :check_status_subject, only: :update
 
+  def show
+    @user_subject = UserSubject.find params[:id]
+    @user_subject.subject.tasks.each do |task|
+      @user_subject.user_tasks.find_or_initialize_by task_id: task.id,
+        user_id: current_user.id
```
Comment (hellovietnam93) [inline app/controllers/subjects_controller.rb]: ident

File: app/views/courses/show.html.erb [hunk 2]
```diff
@@ -1,99 +1,84 @@
...
+              <% if user_subject.finish? %>
+                <div class="status finish">
+              <% elsif user_subject.course_subject.init? %>
+                <div class="status init">
+              <% else %>
+                <div class="status inprogress">
+              <% end %>
+                <strong>
+                  <%= link_to user_subject.subject.name, course_subject_path(@course, user_subject),
+                  {title: user_subject.course_subject.status} %>
```
Comment (hellovietnam93) [inline app/views/courses/show.html.erb]: ident

**Comment by vietanhbkaaa** [comment]
ok

**Comment by hellovietnam93** [comment]
OK

**Pull request merged**